assets/shared/blocks/settings.js

```diff
@@ -1,23 +1,89 @@
+import {
+	ContrastChecker,
+	InspectorControls,
+	PanelColorSettings,
+	withColors,
+} from '@wordpress/block-editor';
+import { __ } from '@wordpress/i18n';
+import { mapValues, upperFirst } from 'lodash';
+
 /**
- * Build PanelColorSettings configuration.
+ * Add color customization support and block settings controls for colors.
  *
- * @param {string}   name                Style attribute name.
- * @param {string}   label               Control label.
- * @param {Object}   props               Component props.
- * @param {Object}   props.style
- * @param {Function} props.setAttributes
- * @return {Object} Configuration.
+ * @param {Object} colorSettings
  */
-export const colorSetting = ( name, label, { style, setAttributes } ) => {
-	return {
-		value: style[ name ],
-		onChange: ( colorValue ) =>
-			setAttributes( {
-				style: {
-					...style,
-					[ name ]: colorValue,
-				},
-			} ),
-		label,
+export function withColorSettings( colorSettings ) {
```

Comment:
In some cases we're using arrow function and others a `function`. Did you use that here for some specific reason? If not, maybe we could update all for the same standard. And maybe we could also add this to our `eslint`. WDYT?

Reply:
Yeah we can do that. Updated in a few places in 888b321